client/src/pages/Booking/Booking.tsx

```diff
@@ -0,0 +1,196 @@
+import { Badge, Box, Card, CardContent, CardHeader, Grid, Paper, Typography } from '@material-ui/core';
+import { DatePicker, MuiPickersUtilsProvider } from '@material-ui/pickers';
+import { MaterialUiPickersDate } from '@material-ui/pickers/typings/date';
+import moment from 'moment';
+import { useEffect, useState } from 'react';
+import NavBar from '../../components/NavBar/NavBar';
+import { listRequests } from '../../helpers/APICalls/listRequests';
+import { Request } from '../../interface/Request';
+import BookingItem from './BookingItem';
+import useStyles from './useStyles';
+
+const Booking = (): JSX.Element => {
+  const [requests, setRequests] = useState<Request[]>([]);
+  const [nextBooking, setNextBooking] = useState<Request[]>([]);
+  const [currentBookings, setCurrentBookings] = useState<Request[]>([]);
+  const [pastBookings, setPastBookings] = useState<Request[]>([]);
+  const [bookingDays, setBookingDays] = useState<string[]>([]);
+  const classes = useStyles();
+  const [anchorEl, setAnchorEl] = useState(null);
+  const [loading, setLoading] = useState<undefined | boolean>(undefined);
+  const [date, setDate] = useState(new Date() as MaterialUiPickersDate);
+
+  const saveRequests = (requests: Request[]) => {
+    setRequests(requests);
+  };
+
+  useEffect(() => {
+    let ignore = true;
+
+    const initBookingDays = (requests: Request[]) => {
+      const dates = requests.map((request) => {
+        return moment(request.start).format('MM/DD/YYYY');
+      });
+      setBookingDays(dates);
+      const nxtBkn: Request[] = [];
+      const crntBkns: Request[] = [];
+      const pstBkns: Request[] = [];
+      const selectedDate = moment(date).startOf('day');
+
+      requests.forEach((booking) => {
+        const startDate = moment(booking.start).startOf('day');
+        if (startDate.diff(selectedDate, 'days') > 0) {
+          nxtBkn.push(booking);
+        } else if (startDate.diff(selectedDate, 'days') === 0) {
+          crntBkns.push(booking);
+        } else {
+          pstBkns.push(booking);
+        }
+      });
+      setNextBooking(nxtBkn);
+      setCurrentBookings(crntBkns);
+      setPastBookings(pstBkns);
+    };
+
+    async function getRequests() {
+      // send request to back end
+      setLoading(true);
+      const response = await listRequests();
+
+      if (ignore && response && response.requests) {
+        saveRequests(response.requests);
+        initBookingDays(response.requests);
+      }
+      setLoading(false);
+    }
+    getRequests();
+    return () => {
+      ignore = false;
+    };
+  }, [date]);
+
+  // eslint-disable-next-line  @typescript-eslint/no-explicit-any
+  const handleClick = (event: any) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const onDateChange = (date: MaterialUiPickersDate): MaterialUiPickersDate => {
+    setDate(date);
+    return date;
+  };
+
+  return (
+    <>
+      <NavBar />
```

Comment:
Maybe consider rendering the Navbar in the 'App.tsx' instead, so that it is persists throughout the app?

Reply:
@fetahokey I think you may have missed implementing the above suggestion 😄